### DISCUSSION TITLE
Ask for new command option "SetOption89 2" for setting MQTT friendly_name instead of short_address - Tasmota Zigbee

### DISCUSSION THREAD

**Discussion by bwu-seevetal** [arendst/Tasmota#19035] [Ideas]
Unexpected from time to time Zigbee devices change their short address, while their friendly name and so all other attributes remain vaild. When setting SetOption89 to "ON", in OPENHAB, the channel of a THING doesn't trigger any more via MQTT, for the short address of the assigned device does not longer exist while changed.
 
It would help, if you could change the MQTT topic part from short address to friendly name - with command "SetOption89 2" for example:
tele/esp32zigbee/8FB2/SENSOR to
tele/esp32zigbee/friendly_name/SENSOR

**Comment by sfromis** [comment]
Try with using [ZbNameTopic](https://tasmota.github.io/docs/Commands/#zbnametopic) (SetOption112)

(BTW. Those SetOption values are binary on/off, cannot support additional values).

**Comment by bwu-seevetal** [comment]
ZbNameTopic does not set individual device topics - it is used to set a topic for all devices of a coordinator.
OPENHAB needa a topic which is individual to each device (which does not change as short addresses do). 
To trigger a channel of a thing in OPENHAB a **topic** is needed - the JSON string is analysed after triggering which does not help.

**Reply by sfromis** [reply]
Doing a test, I see that when I set both ZbDeviceTopic/SetOption89 (which you already use) and ZbNameTopic/SetOption112, the result is that the MQTT topic includes the ZbName value (which is the "friendly name" for Zigbee devices) instead of the short address. That sounds like what you asked for, not so much what you now claim.

**Comment by bwu-seevetal** [comment]
Thanks for help - that is exactly the solution.  Here the documentation lacks a bit,I think ....
